sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -57,12 +60,27 @@ public synchronized void addActivity(final @NotNull Activity activity) {
     if (!isFrameMetricsAggregatorAvailable()) {
       return;
     }
+    snapshotFrameCountsAtStartAndEndPreviousTracking(activity);
     frameMetricsAggregator.add(activity);
-    snapshotFrameCountsAtStart(activity);
+    lastKnownActivity = new WeakReference<>(activity);
   }
 
-  private void snapshotFrameCountsAtStart(final @NotNull Activity activity) {
+  private void snapshotFrameCountsAtStartAndEndPreviousTracking(final @NotNull Activity activity) {
     FrameCounts frameCounts = calculateCurrentFrameCounts();
+    if (lastKnownActivity != null) {
+      @Nullable Activity lastKnownActivityUnwrapped = lastKnownActivity.get();
+      if (lastKnownActivityUnwrapped != null) {
+        @Nullable
+        FrameCounts frameCountDiffOfLastKnownActivity =
+            diffFrameCountsAtEnd(lastKnownActivityUnwrapped, frameCounts);
+        if (frameCountDiffOfLastKnownActivity == null) {
+          frameDiffsAtEnd.put(lastKnownActivityUnwrapped, new FrameCounts(0, 0, 0));
```

Comment:
We fall back to `FrameCounts(0, 0, 0)` so we don't calculate later but freeze it now. This is to avoid double counting.